ext_mongodb.php

```diff
@@ -613,11 +613,90 @@ public function __construct(?array $bulkWriteOptions = array());
 	<<__Native>>
 	public function insert(mixed $document) : mixed;
 
+	private function _queryOptBool( array &$transformedOptions, array $options, string $key )
+	{
+		if ( array_key_exists( $key, $options ) )
+		{
+			$transformedOptions[$key] = (bool) $options[$key];
+		}
+	}
+
+	private function _queryOptDocument( array &$transformedOptions, array $options, string $key )
+	{
+		if ( array_key_exists( $key, $options ) )
+		{
+			if ( !is_array( $options[$key] ) && !is_object( $options[$key] ) )
+			{
+				throw new \MongoDB\Driver\Exception\InvalidArgumentException(
+					sprintf(
+						'Expected "%s" option to be array or object, %s given',
+						$key, gettype( $options[$key] )
+					)
+				);
+			}
+
+			$transformedOptions[$key] = $options[$key];
```

Comment:
PHPC uses `BSON_APPEND_DOCUMENT()` to ensure we always add the converted BSON as a document field. I think the equivalent would be to `(object)` cast here.